phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java

```diff
@@ -0,0 +1,200 @@
+    public static Collection<Boolean> data() {
+        return Arrays.asList(new Boolean[] {
+                true , false});
+    }
+    private String generateDDL(String format) {
+        StringBuilder optionsBuilder = new StringBuilder();
+
+        if (isMultiTenant) {
+            if (optionsBuilder.length() !=0 )
+                optionsBuilder.append(",");
+            optionsBuilder.append("MULTI_TENANT=true");
+        }
+        return String.format(format, isMultiTenant ? "TENANT_ID VARCHAR NOT NULL, " : "",
+                isMultiTenant ? "TENANT_ID, " : "", optionsBuilder.toString());
+    }
+
+    @Test
+    public void testIndexRebuildTask() throws Exception {
+        String baseTable = generateUniqueName();
+        Connection conn = null;
+        Connection viewConn = null;
+        try {
+            conn = DriverManager.getConnection(getUrl());
+            Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+            if (isMultiTenant) {
+                props.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, TENANT1);
+            }
+
+            viewConn =DriverManager.getConnection(getUrl(), props);
+            String ddlFormat =
+                    "CREATE TABLE IF NOT EXISTS " + baseTable + "  ("
+                            + " %s PK2 VARCHAR NOT NULL, V1 VARCHAR, V2 VARCHAR "
+                            + " CONSTRAINT NAME_PK PRIMARY KEY (%s PK2)" + " ) %s";
+            conn.createStatement().execute(generateDDL(ddlFormat));
+            conn.commit();
+            // Create a view
+            String viewName = generateUniqueName();
+            String viewDDL = "CREATE VIEW " + viewName + " AS SELECT * FROM " + baseTable;
+            viewConn.createStatement().execute(viewDDL);
+
+            // Create index
+            String indexName = generateUniqueName();
+            String idxSDDL = String.format("CREATE INDEX %s ON %s (V1)", indexName, viewName);
+
+            viewConn.createStatement().execute(idxSDDL);
+
+            // Insert rows
+            viewConn.createStatement()
+                    .execute(String.format("UPSERT INTO %s VALUES('%s', '%s', '%s')", viewName, "x", "y", "z"));
+            viewConn.commit();
+
+            String data = "{GusWorkId:abc, DisableBefore:true}";
+
+            // Add a task to System.Task to build indexes
+            TaskRegionObserver.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
+                    isMultiTenant? TENANT1 : null, null, viewName, indexName,
+                    PTable.TaskStatus.CREATED.toString(), data, null, null, true);
+
+            // Run IndexRebuildTask
+            TaskRegionObserver.IndexRebuildTask task =
+                    new TaskRegionObserver.IndexRebuildTask(
+                            TaskRegionEnvironment, QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS);
+
+            task.run();
+
+            String viewIndexTableName = "_IDX_" + baseTable;
```

Comment:
nit: should use the appropriate SchemaUtil method to construct the view index table name.